[test][bug] query generator doesn't escape string in REGEXP_LIKE

see:
```
2023-11-01T01:18:33.7042292Z [ERROR]   ExactlyOnceKafkaRealtimeClusterIntegrationTest>BaseRealtimeClusterIntegrationTest.testGeneratedQueries:167->BaseClusterIntegrationTestSet.testGeneratedQueries:488->BaseClusterIntegrationTest.testQuery:738 Caught exception while testing query!
2023-11-01T01:18:33.7046137Z Pinot query: SELECT Quarter, OriginWac, DepTime FROM mytable WHERE  REGEXP_LIKE(OriginCityName, 'Martha'.*s Vineyard, MA') AND Origin > 'MOB' OR OriginAirportID NOT IN (10800, 10581, 10561, 10754, 14635) ORDER BY OriginStateName, OriginAirportSeqID LIMIT 1
2023-11-01T01:18:33.7049870Z H2 query: SELECT `Quarter`, `OriginWac`, `DepTime` FROM mytable WHERE  REGEXP_LIKE(`OriginCityName`, 'Martha'.*s Vineyard, MA', 'i') AND `Origin` > 'MOB' OR `OriginAirportID` NOT IN (10800, 10581, 10561, 10754, 14635) ORDER BY `OriginStateName`, `OriginAirportSeqID` LIMIT 10000
```

in 
```
REGEXP_LIKE(OriginCityName, 'Martha'.*s Vineyard, MA')
```
^ the `Martha'.*s` is weirdly putting a single-quote in the middle and thus result in a wrong SQL.

--- a/pinot-integration-test-base/src/test/java/org/apache/pinot/integration/tests/QueryGenerator.java
+++ b/pinot-integration-test-base/src/test/java/org/apache/pinot/integration/tests/QueryGenerator.java
@@ -1019,7 +1019,8 @@ public class QueryGenerator {
       List<String> columnValues = _columnToValueList.get(columnName);
       String value = pickRandom(columnValues);
       // do regex only for string type
-      if (value.startsWith("'") && value.endsWith("'")) {
+      // do not replace when there's single quote in between b/c it will cause escape issues.
+      if (value.startsWith("'") && value.endsWith("'") && !value.substring(1, value.length() - 1).contains("'")) {
         // replace only one character for now with .* ignore the first and last character
         int indexToReplaceWithRegex = 1 + _random.nextInt(value.length() - 2);
         String regex = value.substring(1, indexToReplaceWithRegex) + ".*" + value.substring(indexToReplaceWithRegex + 1,

